.github/workflows/report-coverage.yml

```diff
@@ -12,6 +12,18 @@ jobs:
     if: github.event.workflow_run.conclusion == 'success'
 
     steps:
+      - name: Set ref for PR
+        if: github.event_name == 'pull_request'
+        run: echo "REF=${{ github.head_ref }}" >> $GITHUB_ENV
+      - name: Set ref for Push
+        if: github.event_name == 'push'
+        run: echo "REF=${{ github.ref }}" >> $GITHUB_ENV
```

Comment:
Is there a reason to prefer this over specifying something like `ref: ${{ github.head_ref || github.ref }}` in the checkout action?